tests/workflows/basic_subworkflow_deployment/tests/test_workflow.py

```diff
@@ -402,3 +402,84 @@ def generate_subworkflow_events(*args: Any, **kwargs: Any) -> Iterator[WorkflowS
     input_names = [input_req.name for input_req in call_kwargs["inputs"]]
     assert "optional_field" not in input_names
     assert len(call_kwargs["inputs"]) == 2
+
+
+def test_stream_workflow__emits_workflow_initiated_and_fulfilled_events(vellum_client):
+    """Confirm that workflow initiated and fulfilled events are emitted and parent context is maintained"""
+
+    # GIVEN a workflow that's set up to hit a Subworkflow Deployment
+    workflow = BasicSubworkflowDeploymentWorkflow()
+
+    # AND we know what the Workflow Deployment will respond with
+    expected_outputs: List[WorkflowOutput] = [
+        WorkflowOutputNumber(id=str(uuid4()), value=70, name="temperature"),
+        WorkflowOutputString(
+            id=str(uuid4()), value="I went to weather.com and looked at today's forecast.", name="reasoning"
+        ),
+    ]
+
+    execution_id = str(uuid4())
+    expected_events: List[WorkflowStreamEvent] = [
+        WorkflowExecutionWorkflowResultEvent(
+            execution_id=execution_id,
+            data=WorkflowResultEvent(
+                id=str(uuid4()),
+                state="INITIATED",
+                ts=datetime.now(),
+            ),
+        ),
+        WorkflowExecutionWorkflowResultEvent(
+            execution_id=execution_id,
+            data=WorkflowResultEvent(
+                id=str(uuid4()),
+                state="FULFILLED",
+                ts=datetime.now(),
+                outputs=expected_outputs,
+            ),
+        ),
+    ]
+
+    def generate_subworkflow_events(*args: Any, **kwargs: Any) -> Iterator[WorkflowStreamEvent]:
+        yield from expected_events
+
+    vellum_client.execute_workflow_stream.side_effect = generate_subworkflow_events
```

Comment:
During subworkflow deployment inlining, we don't call the execute workflow stream method, but rather, vellum pull gets called so that the workflow is run locally (or is already there from the outset of the method). I would take a look at the impl of `_run_resolved_workflow`